### PR TITLE
Use Transform Dialect with matmul

### PR DESCRIPTION
**Pull request by JonasCrols**
Add kernel test where simple_matmul script gets transformed using transform dialect to become half_tiled_matmul

### CODE REVIEW

File: runtime/Makefile.rules [hunk 1]
```diff
@@ -74,9 +74,16 @@ LDFLAGS += -lsnRuntime
...
+
+MLIRTRANSFORM = --pass-pipeline='builtin.module(test-transform-dialect-interpreter{bind-first-extra-to-ops=linalg.quantized_matmul}, test-transform-dialect-erase-schedule)'
+
+%.mlir: %.transform.mlir
+	$(MLIROPT) $(MLIRTRANSFORM) -o $@ $<
+
 # MLIR Preprocessing
 
-MLIRPREPROCFLAGS = --pass-pipeline='builtin.module(func.func(tosa-to-linalg-named, tosa-to-tensor, tosa-to-scf, tosa-to-linalg))'
+MLIRPREPROCFLAGS = --pass-pipeline='builtin.module( func.func(tosa-to-linalg-named, tosa-to-tensor, tosa-to-scf, tosa-to-linalg))'
```
Comment (JosseVanDelm) [inline runtime/Makefile.rules:86]: Not sure why this change is necessary.

File: kernels/transform_matmul/matmul.transform.mlir [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+transform.sequence failures(propagate) {
+^bb0(%arg0: !transform.any_op,
+     %arg1: !transform.op<"linalg.quantized_matmul">):
+  // The actual tiling transformation takes tile sizes as attributes.
+   %loop1, %loop2, %tiled = transform.structured.tile %arg1 [8, 8]
+    : (!transform.op<"linalg.quantized_matmul">) -> (!transform.any_op, !transform.any_op, !transform.any_op)
+  transform.yield
+}
+
```
Comment (JosseVanDelm) [inline kernels/transform_matmul/matmul.transform.mlir:16]: If this is the only line that is changing alongside the added options to run the compiler, we might just want to add this to another folder instead, of adding yet another entire 200 lines. @jorendumoulin what do you think?

**Review by JosseVanDelm — APPROVED** [review]
Very nice :heart: :heart_eyes_cat: :heart_hands:

File: kernels/transform_matmul/Makefile [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Courtesy of Federico Ficarelli
+
+.DEFAULT_GOAL := all
+
+include ../../runtime/snax-gemm.rules
```
Comment (jorendumoulin) [inline kernels/transform_matmul/Makefile:5]: Let's use the streamer gemm?

**Review by jorendumoulin — APPROVED** [review]
Cool!

I agree with Josse - Let's just put this in the `streamer_gemm` folder. Only the mlir file is different I think?
